### PR TITLE
Improve examples

### PR DESCRIPTION
**Pull request by BrynCooke**
Change the names of some of the example directories
Add README.md to each example
Rename config files.

### CODE REVIEW

File: examples/async-auth/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        ServiceBuilder::new()
+            .checkpoint_async(...) // Authentication happens here 
+            .buffer(20_000) // Required, see note below
+            .service(service)
+            .boxed()
+    }
+```
+
+Note that layers that require a service to me moved across an `await` point, e.g. `checkpoint_async` or `filter_async`
+must be followed by a call to buffer, as they require the downstream service to be `Clone`.
```
Comment (garypen) [inline examples/async-auth/README.md:23]:
```suggestion
Note that layers that require a service to be moved across an `await` point, e.g. `checkpoint_async` or `filter_async`
must be followed by a call to buffer, as they require the downstream service to be `Clone`.
```

File: examples/context/router.yaml [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
 server:
   cors:
     origins:
       - "https://studio.apollographql.com/"
+telemetry:
+  opentelemetry:
+    jaeger:
+
+override_subgraph_url:
+
```
Comment (garypen) [inline examples/context/router.yaml:11]: Should this be part of this configuration?

File: examples/hello-world/src/main.rs [hunk 3]
```diff
@@ -12,7 +12,7 @@ use anyhow::Result;
 // you can use the plugin by adding it to `config.yml`
 mod hello_world;
 
-// `cargo run -- -s ../supergraph.graphql -c ./config.yml`
+// `cargo run -- -s ../supergraph.graphql -c ./router.yaml`
```
Comment (garypen) [inline examples/hello-world/src/main.rs:15]:
```suggestion
// `cargo run -- -s ../graphql/supergraph.graphql -c ./router.yaml`
```

File: examples/jwt-auth/README.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# JWT authentication
+
+DISCLAIMER: This is an example for illustrative purposes. It has not been security audited and is purely intended of an 
+illustration of an approach to JWT verification via a router plugin.
```
Comment (garypen) [inline examples/jwt-auth/README.md:4]:
```suggestion
DISCLAIMER: This is an example for illustrative purposes. It has not been security audited and is purely intended as an 
illustration of an approach to JWT verification via a router plugin.
```

File: examples/jwt-auth/src/main.rs [hunk 5]
```diff
@@ -19,7 +19,7 @@ use anyhow::Result;
 // you can use the plugin by adding it to `config.yml`
 mod jwt;
 
-// `cargo run -- -s ../supergraph.graphql -c ./config.yml`
+// `cargo run -- -s ../supergraph.graphql -c ./router.yaml`
```
Comment (garypen) [inline examples/jwt-auth/src/main.rs:22]:
```suggestion
// `cargo run -- -s ../graphql/supergraph.graphql -c ./router.yaml`
```

**Review by garypen — CHANGES_REQUESTED** [review]
Some small edits.

File: examples/context/router.yaml (hunk 2 above)
Comment (BrynCooke) [reply examples/context/router.yaml:11]: Oops

**Pull request closed without merging**